05.ls/ls05.rb

```diff
@@ -8,14 +8,26 @@
 ROW_MAX_WIDTH = 24
 
 def main
-  all_files = Dir.glob('*').sort
-  opt = OptionParser.new
-  opt.on('-l')
-  opt.parse(ARGV)
-  if ARGV == ['-l']
+  params = ARGV.getopts('alr')
+  all_files = get_all_files(params)
+  get_result(params, all_files)
+end
+
+def get_all_files(params)
+  files = if params['a']
+            Dir.glob('*', File::FNM_DOTMATCH).sort
+          else
+            Dir.glob('*').sort
+          end
```

Comment:
このままでも全く問題はないんですが、条件文もシンプルですし、if/elseの中身も一行なので、こういう場合は三項演算子を用いて書くと行数が短くなって良い…かもしれません。好みの問題もありますのでお好きな方でどうぞ。

```suggestion
  files = params['a'] ? Dir.glob('*', File::FNM_DOTMATCH).sort : Dir.glob('*').sort
```